src/RomaricDrigon/MetaYaml/Exception/NodeValidatorException.php

```diff
@@ -5,16 +5,50 @@
 class NodeValidatorException extends \Exception
 {
     private $node_path;
+    private $paths = [];
+    private $messages = [];
 
-    public function __construct($node_path, $message)
+  public function __construct($node_path, $messages, $paths)
```

Comment:
I'm not sure of why we need to pass an array of messages? The code is much less visible and I'm not sure `current($messages)` will always return what we expect. What about having `$message` (string) and an extra parameter `$previousMessages`(array) ?

Reply:
And it looks like there's a small indentation glitch just in front of the constructor.